nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -783,10 +783,11 @@ def loss_func(self, loss_mask, tokens_loss):
         """
         This function takes as input per-token loss and masks non-required values.
         """
-        losses = tokens_loss.view(-1).float()
-        loss_mask = loss_mask.view(-1).float()
-        # TODO: add nemo version here
-        loss = torch.sum(losses * loss_mask) / loss_mask.sum()  # sequence level nll
+        losses = tokens_loss.float()
```

Comment:
Can you add more verbose doc string? It would be nice to know what shapes are expected here and which dimension correspond to what.

Reply:
Indeed, this is always a good idea => done in 97e9d05486042b586e121b3fd21751d0fe452041

Reply:
I hoped for something more verbose :) Like
"""
This function takes as input per-token loss and masks non-required values.
Input:
    loss_mask: binary torch.Tensor of shape batch_size x seq_len
    loss_tokens: torch.Tensor of shape batch_size x seq_len
Returns:
    loss: 0-dimension torch.Tensor (single value)
"""

Reply:
There isn't a single loss function in NeMo (that I can see) that has such a detailed docstring. I'm not saying it wouldn't be a good idea to add them, but this is out-of-scope for this PR which is focused on fixing a bug. It seems to me that adding shapes in comments should be enough to follow the loss calculation.